pkg/operator/ceph/object/spec.go

```diff
@@ -872,10 +872,10 @@ func (c *clusterConfig) generateVolumeSourceWithCaBundleSecret() (*v1.SecretVolu
 
 func (c *clusterConfig) rgwTLSSecretType(secretName string) (v1.SecretType, error) {
 	rgwTlsSecret, err := c.context.Clientset.CoreV1().Secrets(c.clusterInfo.Namespace).Get(c.clusterInfo.Context, secretName, metav1.GetOptions{})
-	if rgwTlsSecret != nil {
-		return rgwTlsSecret.Type, nil
+	if err != nil {
+		return "", errors.Wrapf(err, "no Kubernetes secrets referring TLS certificates found")
```

Comment:
```suggestion
		return "", errors.Wrapf(err, "failed to get Kubernetes secrets referring the TLS certificates")
```

NIT. Current error message `no Kubernetes secrets referring TLS certificates found` is very specifically states that the secret is not present. It could be a different error, so this error message would be confusing. So suggesting a more generic message.